### ISSUE TITLE
I can build a road that is not connecting

### ISSUE THREAD

**Issue by yeyyosing** [thyer/CS340winter2016#545]
I can build  road that doesn't connect because it passes through the settlement of another player.  If the settlement wasn't there, then the road would be considered connected.  Since the settlement of another player is there however, it is not considered connected and I shouldn't be allowed to build it.

<img width="175" alt="screen shot 2016-03-25 at 4 29 51 pm" src="https://cloud.githubusercontent.com/assets/16694151/14055387/1d0fb6be-f2a7-11e5-82af-80d5a926877f.png">

**Comment by matthewfcarlson** [comment]
Valid bug. Sorry we didn't know that was a rule.

**Comment by yeyyosing** [comment]
quick question, did you guys implement the longest road algorithm?

**Comment by matthewfcarlson** [comment]
Yeah we did. It would be affected by this.

**Comment by yeyyosing** [comment]
Ok because the longest road didn't update either but I think it is affected by this bug

**Comment by sadler-j** [comment]
Yeah, I guess I need to recheck the rules in that regard. So the first bug is definitely an issue since I misunderstood the rules. How long was the road you built? I did implement longest road and it was working properly when I tested it, at least for the cases I tested.

**Comment by yeyyosing** [comment]
i'll try to reproduce the bug if I can, but it may stem from the first bug.  I'll keep on testing it

**Comment by yeyyosing** [comment]
<img width="682" alt="screen shot 2016-03-25 at 5 20 14 pm" src="https://cloud.githubusercontent.com/assets/16694151/14056153/fa8fb0f6-f2ad-11e5-82f0-b274ef2fb1cf.png">

So do you notice here that white has the longest road icon? that means it would take into account other buildings cutting off connecting roads, because red has a longer road than white, yet it has building cutting it off.  However, I can build a road through a building, so it doesn't look like that would be taken into account, and thus red would have the longest road.  I don't know if this is one or two bugs because they are mixed up in this case.  Just look into it

**Comment by sadler-j** [comment]
White should have the longest road in this case as red's road is three roads of length 3, 2, and 4 respectively, so that isn't a bug. The other issue is if red's roads should be allowed if they were built after the yellow and green settlements. Of they shouldn't have been allowed, then that's a bug, but I need to check the rules on that. That is where I may be mistaken on the rules. The longest road takes into account the settlements where placing roads don't care. At the moment, placing roads only looks to see if there is a settlement or road at one of the adjacent vertices.
